ntp.go

```diff
@@ -173,27 +189,49 @@ func Query(host string, version int) (*Response, error) {
 }
 
 // getTime returns the "receive time" from the remote NTP server host.
-func getTime(host string, version int) (*msg, error) {
-	if version < 2 || version > 4 {
+func getTime(host string, opt QOption) (*msg, error) {
+	if opt.Version == 0 {
+		opt.Version = defaultNtpVersion
+	}
+
+	if opt.Version < 2 || opt.Version > 4 {
 		panic("ntp: invalid version number")
 	}
 
+	if opt.Timeout == 0 {
+		opt.Timeout = defaultTimeout
+	}
+
 	raddr, err := net.ResolveUDPAddr("udp", host+":123")
 	if err != nil {
 		return nil, err
 	}
 
+	if opt.Port != 0 {
+		raddr.Port = opt.Port
+	}
+
 	con, err := net.DialUDP("udp", nil, raddr)
 	if err != nil {
 		return nil, err
 	}
 	defer con.Close()
-	con.SetDeadline(time.Now().Add(timeout))
+
+	if opt.IpTTL != 0 {
+		ipcon := ipv4.NewConn(con)
```

Comment:
I'm not familiar with the `golang.org/x/net/ipv4` package.  Is this all you have to do to set a TTL on packets sent through the `con`?  Are you sure you don't have to declare the `ipcon` outside the `if` statement and use it instead of `con` in the `binary.Write` call below?